### ISSUE TITLE
affineFormats

### ISSUE THREAD

**Issue by Azshar** [react-native-text-input-mask/react-native-text-input-mask#217]
<img width="482" alt="Снимок экрана 2021-04-16 в 19 36 27" src="https://user-images.githubusercontent.com/20275419/115056398-76a59b80-9eeb-11eb-8ea5-9d8555437ad4.png">

**Comment by fuadmahmud** [comment]
I thought it was just me having the problem, have done doing cleanup npm, reinstall the module and manual linking, yet still have this issue.

**Comment by hulohot** [comment]
I'm also having this issue.

**Comment by jeremyfall80** [comment]
I posted a question on Stackoverflow.
https://stackoverflow.com/questions/67320406/how-can-i-fix-react-native-android-affineformats-error

**Comment by castor55** [comment]
I posted a question on Stackoverflow.
https://stackoverflow.com/questions/67320406/how-can-i-fix-react-native-android-affineformats-error

**Comment by castor55** [comment]
@martintreurnicht Could you check this issue soon?

**Comment by limdev92** [comment]
The error disappear using the version v3.0.4

**Comment by castor55** [comment]
So if once you installed v3.1.4, please remove it first and add `react-native-text-input-mask": "3.0.4` in package.json file.
You need to run `yarn` after that.

**Comment by castor55** [comment]
@Azshar You can close this issue.

**Comment by limdev92** [comment]
> @Azshar You can close this issue.

wait, this is not solved, the downgrade of the version is not a solution

**Comment by castor55** [comment]
> > @Azshar You can close this issue.
> 
> wait, this is not solved, the downgrade of the version is not a solution

Agree. Btw, I see another issue. It does not compile properly in iOS.

**Comment by castor55** [comment]
v3.0.4 does not work on iOS 12.4 with the following package versions.
"react": "16.13.1",
"react-native": "0.63.3",
"react-native-text-input-mask": "3.0.4"

```
The following build commands failed:
        CompileSwiftSources normal x86_64 com.apple.xcode.tools.swift.compiler
        CompileSwift normal x86_64 /Volumes/DATA/Project/EZ/React-Native/node_modules/react-native-text-input-mask/ios/TextInputMask.swift
(2 failures)
```

**Comment by castor55** [comment]
So this issue should be fixed on v3.1.4.

**Comment by castor55** [comment]
I figure out this issue by fix the pod file. We need to disable Flipper.
  ```
use_frameworks!
  config = use_native_modules!

  use_react_native!(:path => config["reactNativePath"])

  pod 'GoogleSignIn', '~> 5.0.2'

  $static_framework = ['FlipperKit', 'Flipper', 'Flipper-Folly',
    'CocoaAsyncSocket', 'ComponentKit', 'DoubleConversion',
    'glog', 'Flipper-PeerTalk', 'Flipper-RSocket', 'Yoga', 'YogaKit',
    'CocoaLibEvent', 'OpenSSL-Static', 'boost-for-react-native']

  pre_install do |installer|
    Pod::Installer::Xcode::TargetValidator.send(:define_method, :verify_no_static_framework_transitive_dependencies) {}
    installer.pod_targets.each do |pod|
      if $static_framework.include?(pod.name)
        pod.instance_variable_set(:@host_requires_frameworks, false)
        puts pod
      end
    end
  end
```

**Comment by luisfcerutti** [comment]
A workaround is setting in the input mask the properties it misses, example:

<TextInputMask
                    onChangeText={onChangeText}
                    value={value}
                    affineFormats={[]}
                    customNotations={[]}
                    affinityCalculationStrategy={'WHOLE_STRING'}
                    mask={"[000].[000].[000]"}
 />

it worked for me

**Comment by castor55** [comment]
v3.0.4 has pod issues in iOS.

**Comment by castor55** [comment]
@luisfcerutti Were you able to run the Android app without seeing `affineFormats` issue? Are you using v3.1.4?

**Comment by kopax-polyconseil** [comment]
Thanks @luisfcerutti I was about to uninstall that module.

**Comment by martintreurnicht** [comment]
it's unclear to me what the issue is here. `affineFormats` do work, I use that in my own project. So just an issue with cocoapods? You can use it without frameworks by the way, see installation instruction as static library. @Azshar can this issue be closed now?
